### PR TITLE
only return streams that are exact

### PR DESCRIPTION
**Pull request by binarymatt**

### CODE REVIEW

File: logstream.go [hunk 1]
```diff
@@ -73,6 +73,10 @@ func (s *LogStream) findStream() (*cloudwatchlogs.LogStream, error) {
 		return nil, nil
 	}
 
+	if *resp.LogStreams[0].LogStreamName != *s.Stream {
```
Comment (bradgignac) [inline logstream.go]: Does the limit need to be removed so we can check all results in the `DescribeLogStreamsInput` response?

**Comment by binarymatt** [comment]
@bradgignac what about instead of looking for an exact match, changing the way streams are named so that there is a terminator behind the ip-address?

**Comment by bradgignac** [comment]
@binarydud I'd rather not impose that constraint.

File: logstream.go [hunk 2]
```diff
@@ -69,15 +78,15 @@ func (s *LogStream) findStream() (*cloudwatchlogs.LogStream, error) {
 		return nil, err
 	}
 
-	if len(resp.LogStreams) == 0 {
-		return nil, nil
-	}
+	streams := Filter(resp.LogStreams, func(st *cloudwatchlogs.LogStream) bool {
```
Comment (bradgignac) [inline logstream.go]: `Filter` is great for the general case, but maybe an `any` method makes more sense for this case?

```
func any(vs []*cloudwatchlogs.LogStream, f func(*cloudwatchlogs.LogStream) bool) []*cloudwatchlogs.LogStream {
    for _, v := range vs {
        if f(v) {
            return true
        }
    }
    return false
}
```

File: logstream.go [hunk 3]
```diff
@@ -17,6 +17,15 @@ type LogStream struct {
 	service *cloudwatchlogs.CloudWatchLogs
 }
 
+func Filter(vs []*cloudwatchlogs.LogStream, f func(*cloudwatchlogs.LogStream) bool) []*cloudwatchlogs.LogStream {
```
Comment (bradgignac) [inline logstream.go]: This should probably be lowercase as we don't want to export it from the package.

Comment (binarymatt) [inline logstream.go]: thanks, i forgot about that part of go

**Comment by binarymatt** [comment]
@bradgignac changed the name to be private.

**Comment by binarymatt** [comment]
@bradgignac i went with filter over any because of the need to return the stream. Any tells me if a stream existing stream matches. It doesn't tell me which stream matches.

**Comment by bradgignac** [comment]
:shipit:

**Pull request merged**